### PR TITLE
Add bfloat16 support for lerp on CPU

### PR DESCRIPTION
**Pull request by CaoE**
### Description
Add bfloat16 support for lerp on CPU

### Testing
single core:
<html>
<body>

op | shape |fp32 forward/ms|bf16 forward/s|fb32 backward/s| bf16 backward/s
-- | -- | -- | -- | -- | --
lerp (tensor) | [10, 128, 10, 124] | 0.005489 | 0.000613 | 0.006658 | 0.003385
  | [10, 128, 20, 124] | 0.011057 | 0.001204 | 0.016032 | 0.007869
  | [10, 128, 30, 124] | 0.016691 | 0.001954 | 0.025549 | 0.012823
  |   |   |   |   |  
lerp (scalar) | [10, 128, 10, 124] | 0.001096 | 0.000507 | 0.002024 | 0.001479
  | [10, 128, 20, 124] | 0.00247 | 0.000997 | 0.005468 | 0.002907
  | [10, 128, 30, 124] | 0.004178 | 0.001513 | 0.009775 | 0.004859

</body>
</html>

single socket (28cores):
<html>
<body>

op | shape | fp32 forward/s| bf16 forward/s| fb32backward/s| bf16 backward/s
-- | -- | -- | -- | -- | --
lerp (tensor) | [10, 128, 10, 124] | 0.000236 | 3.93E-05 | 0.000494 | 0.000235
  | [10, 128, 20, 124] | 0.000525 | 7.39E-05 | 0.002485 | 0.000638
  | [10, 128, 30, 124] | 0.000801 | 0.000121 | 0.004235 | 0.001529
  |   |   |   |   |  
lerp (scalar) | [10, 128, 10, 124] | 5.90E-05 | 3.32E-05 | 0.000129 | 0.000116
  | [10, 128, 20, 124] | 0.000155 | 5.87E-05 | 0.000368 | 0.000206
  | [10, 128, 30, 124] | 0.000324 | 9.04E-05 | 0.001322 | 0.000313

</body>
</html>

### CODE REVIEW

**Comment by mruberry** [comment]
@ngimel do you know the best bfloat16 cpu reviewer?

**Comment by frank-wei** [comment]
> @ngimel do you know the best bfloat16 cpu reviewer?

Similar question here and it is good to know the best person for this feature. Currently, I am a temp reviewer for this feature.

**Comment by ngimel** [comment]
I thought @frank-wei is the reviewer for bfloat16-related PRs.

File: aten/src/ATen/native/cpu/LerpKernel.cpp [hunk 1]
```diff
@@ -9,7 +9,36 @@ namespace native {
...
+    using bVec = Vectorized<BFloat16>;
+    using fVec = Vectorized<float>;
+    float weight_val = weight.to<float>();
+    auto weight_vec = fVec(weight_val);
+    auto threshold_vec = fVec(0.5);
+    auto one_vec = fVec(1);
+    at::native::cpu_kernel_vec(
+      iter,
+      [weight_val](BFloat16 self_val, BFloat16 end_val) -> BFloat16 {
+        return (weight_val < 0.5)
```
Comment (frank-wei) [inline aten/src/ATen/native/cpu/LerpKernel.cpp:22]: Generally looks good to me. The bfloat16 is cast to float for scalar operation. It should have similar perf as float. But why in your perf table, the scalar bfloat16 is much faster than scalar float?

Comment (CaoE) [reply aten/src/ATen/native/cpu/LerpKernel.cpp:22]: Because bfloat16 uses cpu_kernel_vec and blendv, which is faster than cpu_kernel.

**Review by frank-wei — APPROVED** [review]
@pytorchbot merge -l

**Pull request closed without merging**